### PR TITLE
Provide a switch to disable Spring DI bean creation

### LINKED ISSUE #35023: Add build-time property to disable `quarkus-spring-di`
### Description

As discussed [here (and the subsequent comments)](https://github.com/quarkusio/quarkus/issues/35011#issuecomment-1651260879), it can make sense to disable `quarkus-spring-di`, e.g. when you just want `spring-data-jpa` features without any spring injection support (which might increase the risk of issues/side effects, like #35011).

### Implementation ideas

_No response_

### PR DESCRIPTION
**Pull request by geoand**
Closes: #35023

### CODE REVIEW

File: extensions/vertx-http/runtime/src/main/java/io/quarkus/vertx/http/runtime/StaticResourcesRecorder.java
Comment (famod) [inline extensions/vertx-http/runtime/src/main/java/io/quarkus/vertx/http/runtime/StaticResourcesRecorder.java:1]: This change looks unrelated?

File: extensions/spring-di/deployment/src/main/java/io/quarkus/spring/di/deployment/SpringDiBuildTimeConfig.java [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+import io.smallrye.config.WithDefault;
+
+@ConfigMapping(prefix = "quarkus.spring-di")
+@ConfigRoot(phase = ConfigPhase.BUILD_TIME)
+public interface SpringDiBuildTimeConfig {
+
+    /**
+     * Whether Spring DI is enabled **during the build**.
+     * <p>
+     * Turning this setting off will result in Quarkus completely ignoring beans annotated with Spring annotations
```
Comment (famod) [inline extensions/spring-di/deployment/src/main/java/io/quarkus/spring/di/deployment/SpringDiBuildTimeConfig.java:15]:
```suggestion
     * Turning this setting off will result in Quarkus completely ignoring beans annotated with Spring annotations to reduce possible side-effects when Spring DI is brought in transitively (e.g. via quarkus-spring-data-jpa).
```
Maybe?

**Comment by famod** [comment]
I'll test this without #35015

File: extensions/vertx-http/runtime/src/main/java/io/quarkus/vertx/http/runtime/StaticResourcesRecorder.java
Comment (geoand) [reply extensions/vertx-http/runtime/src/main/java/io/quarkus/vertx/http/runtime/StaticResourcesRecorder.java:1]: Oops!

File: extensions/spring-di/deployment/src/main/java/io/quarkus/spring/di/deployment/SpringDiBuildTimeConfig.java (hunk 1 above)
Comment (geoand) [reply extensions/spring-di/deployment/src/main/java/io/quarkus/spring/di/deployment/SpringDiBuildTimeConfig.java:15]: I don't think we need to be specific to be honest

Comment (famod) [reply extensions/spring-di/deployment/src/main/java/io/quarkus/spring/di/deployment/SpringDiBuildTimeConfig.java:15]: Alright, I was a bit on the fence myself.

**Review by famod — APPROVED** [review]
Works as expected, thanks!
